Add CLI version switching

jribbink opened the pull request:
Closes #538 

## Description

Adds ability to select versions of the CLI with native support for `flow-c1`

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

jribbink commented:
tests pass locally - there are seemingly issues with `@vscode/test-electron` that break these, will be addressed subsequently (probably need to update)

bthaile commented on extension/src/server/flow-config.ts:
```diff
@@ -172,7 +172,7 @@ export class FlowConfig implements Disposable {
   // Watch and reload flow configuration when changed.
   #watchWorkspaceConfiguration (): Subscription {
-    return this.#settings.watch$(config => config.customConfigPath).subscribe(() => {
+    return this.#settings.settings$(config => config.customConfigPath).subscribe(() => {
```
settings.settings ? this look weird, just pointing it out if it's a typo

jribbink replied:
Yeah, I'll switch the back I think.  I was trying to follow the pattern with the other observables where we have `XX.value$` for the observable and `XX.getValue()` to get the current value, but it's probably a bit confusing for settings.

bthaile reviewed (approved):
lgtm

bthaile commented on extension/src/server/language-server.ts:
```diff
@@ -3,34 +3,37 @@ import { window } from 'vscode'
 
     this.#isActive = true
-    await this.startClient()
+    await this.startClient().catch(() => {})
```
would an error message here get swelled?

jribbink replied:
Yeah good callout.  I'll fix this & add comments as to why exactly this error is caught.

bthaile commented:
How does the correct language server get chosen based on the version of flow-cli used?

jribbink commented:
> How does the correct language server get chosen based on the version of flow-cli used?

The `LanguageServerAPI` watches for the binary to change from the `CliProvider` and will restart accordingly.  I realized it maybe wasn't super clear, so I've moved some stuff around and added some more comments to make it more obvious that this is how it works.
